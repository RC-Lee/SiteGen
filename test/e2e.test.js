const { run } = require('./run');

describe('end-to-end integration', () => {
  test('prints error and help message when no arguments given', async () => {
    const { stderr, stdout, exitCode } = await run();
    const errorMsg = 'error: required option input <file or directory> is not specified';
    expect(exitCode === 0).toBeFalsy();
    expect(stderr.includes(errorMsg)).toBeTruthy();
    expect(stdout).toEqual('');
  });

  test('prints help message when --help given', async () => {
    const { stderr, stdout, exitCode } = await run('--help');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('prints version message when --version given', async () => {
    const { stderr, stdout, exitCode } = await run('--version');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('single valid text file input should generate an html', async () => {
    const { stderr, stdout, exitCode } = await run('--input', './test/sample.txt');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('single valid text file input should generate an html', async () => {
    const { stderr, stdout, exitCode } = await run(
      '--input',
      './test/sample.txt',
      '--stylesheet',
      'link'
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('single valid text file input with valid output folder should generate an html in said folder', async () => {
    const { stderr, stdout, exitCode } = await run(
      '--input',
      './test/sample.txt',
      '--output',
      './test/outputTest'
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('single valid text file input with invalid output folder should generate an html in ./dist', async () => {
    const { stderr, stdout, exitCode } = await run(
      '--input',
      './test/sample.txt',
      '--output',
      './test/invalid'
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });
});
