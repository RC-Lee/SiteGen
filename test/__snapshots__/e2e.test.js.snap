// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints help message when --help given 1`] = `
"Usage: index [options]

Options:
  -v, --version                      Output the current version
  -i, --input <file or directory>    Designate an input file or directory
  -o, --output <directory>           Designate an ouput directory
  -s, --stylesheet <stylesheet url>  Link to a stylesheet url (default:
                                     \\"https://cdn.jsdelivr.net/npm/water.css@2/out/water.css\\")
  -c, --config <config json file>    Link to file that specifies all SSG
                                     options
  -h, --help                         display help for command"
`;

exports[`end-to-end integration prints version message when --version given 1`] = `
"Name: sitegen 
Version: 0.1.0"
`;

exports[`end-to-end integration single valid text file input should generate an html 1`] = `""`;

exports[`end-to-end integration single valid text file input with invalid output folder should generate an html in ./dist 1`] = `"Output directory ./test/invalid doesn't exist, outputting all files to ./dist"`;

exports[`end-to-end integration single valid text file input with valid output folder should generate an html in said folder 1`] = `""`;
